### PR TITLE
Add more tests for DateTimeFormat.hourCycle/hour12.

### PR DESCRIPTION
**Pull request by Ms2ger**
In particular, when using dateStyle/timeStyle.

See also <https://github.com/tc39/proposal-intl-datetime-style/issues/34>.

### CODE REVIEW

File: test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                   `Should support dateStyle=${dateStyle}`);
+
+  /* Values passed via unicode extension key work */
+
+  for (const hcValue of hcValues) {
+    const resolvedOptions = new Intl.DateTimeFormat(`de-u-hc-${hcValue}`, {
+      dateStyle,
+    }).resolvedOptions();
+
+    assert.sameValue(resolvedOptions.hourCycle, undefined);
```
Comment (FrankYFTang) [inline test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js]: Why is this undefined? 
https://tc39.es/proposal-intl-datetime-style/#sec-initializedatetimeformat
```
21. Set dateTimeFormat.[[HourCycle]] to hc.
```

File: test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    assert.sameValue(resolvedOptions.hour12, undefined);
```
Comment (FrankYFTang) [inline test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js]: depending on above.

File: test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  /* Values passed via options work */
+
+  for (const hcValue of hcValues) {
+    const resolvedOptions = new Intl.DateTimeFormat("en-US", {
+      dateStyle,
+      hourCycle: hcValue
+    }).resolvedOptions();
+
+    assert.sameValue(resolvedOptions.hourCycle, undefined);
+    assert.sameValue(resolvedOptions.hour12, undefined);
```
Comment (FrankYFTang) [inline test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js]: same issue as above. Why are these undefined?

File: test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  assert.sameValue(resolvedOptions.hour12, undefined);
+
+  resolvedOptions = new Intl.DateTimeFormat("fr", {
+    dateStyle,
+    hour12: true,
+    hourCycle: "h23"
+  }).resolvedOptions();
+
+  assert.sameValue(resolvedOptions.hourCycle, undefined);
+  assert.sameValue(resolvedOptions.hour12, undefined);
```
Comment (FrankYFTang) [inline test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js]: "21. Set dateTimeFormat.[[HourCycle]] to hc." 
so why are these undefined?

File: test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+  resolvedOptions = new Intl.DateTimeFormat("fr-u-hc-h24", {
+    dateStyle,
+    hour12: true,
+  }).resolvedOptions();
+
+  assert.sameValue(resolvedOptions.hourCycle, undefined);
```
Comment (FrankYFTang) [inline test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js]: same here

**Comment by Ms2ger** [comment]
Note the accompanying spec change in https://github.com/tc39/proposal-intl-datetime-style/issues/34

**Comment by FrankYFTang** [comment]
ok, with  tc39/proposal-intl-datetime-style#34 it make sense. LGTM after #34 merged in.

**Comment by leobalter** [comment]
Just for future reference...

Depends on:

- [ ]  tc39/proposal-intl-datetime-style#34 
- [ ]  tc39/proposal-intl-datetime-style#37

File: test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js (hunk 3 above)
Comment (FrankYFTang) [reply test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js]: the question is answered. No longer issue. Please approve/merge this PR

File: test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js (hunk 4 above)
Comment (FrankYFTang) [reply test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js]: the question is answered. No longer issue. Please approve/merge this PR

File: test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js (hunk 5 above)
Comment (FrankYFTang) [reply test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js]: the question is answered. No longer issue. Please approve/merge this PR

**Comment by FrankYFTang** [comment]
Dependency are all MERGED/CLOSED. please merge this PR

**Pull request merged**